### ISSUE TITLE
Label logic needs separating from metrics

### ISSUE THREAD

**Issue by neworderofjamie** [genn-team/ml_genn#55]
- Multiple metrics e.g. ``SparseCategoricalAccuracy`` and ``SparseCategoricalAccuracy`` require access to ``YTrue``
- Would also allow compilers to add stuff like ``YTrueBack`` more generically
- However, in Keras, I _think_ label dimensionality does come from metrics/loss functions
